packages/cosmwasm-stargate/src/cosmwasmclient.ts

```diff
@@ -240,10 +251,15 @@ export class CosmWasmClient {
         : pollForTx(txId);
     };
 
+    const broadcasted = await this.forceGetTmClient().broadcastTxSync({ tx });
+    if (broadcasted.code) {
+      throw new Error(
+        `Broadcasting transaction failed with code ${broadcasted.code} (codespace: ${broadcasted.codeSpace}). Log: ${broadcasted.log}`,
+      );
+    }
+    const transactionId = toHex(broadcasted.hash).toUpperCase();
     return new Promise((resolve, reject) =>
-      this.forceGetTmClient()
-        .broadcastTxSync({ tx })
-        .then(({ hash }) => pollForTx(toHex(hash).toUpperCase()))
+      pollForTx(transactionId)
```

Comment:
I'm still unsure if the use of

```
        .then(resolve, reject)
        .finally(() => clearTimeout(txPollTimeout)),
```

is correct. Shoudn't the call of `resolve` or `resolve` be the last thing that happens here?

Reply:
Happy to move the `clearTimeout` into the `then` handlers to make it clearer.

Reply:
Done in 33ff475. This changes the order of execution such that nothing happens after the resolve/reject call.